Python/sysmodule.c

```diff
@@ -1793,7 +1789,13 @@ sys__getframe_impl(PyObject *module, int depth)
                          "call stack is not deep enough");
         return NULL;
     }
-    return _Py_XNewRef((PyObject *)_PyFrame_GetFrameObject(frame));
+
+    PyObject *pyFrame = _Py_XNewRef((PyObject *)_PyFrame_GetFrameObject(frame));
+    if (_PySys_Audit(tstate, "sys._getframe", "(O)", pyFrame) < 0) {
+        Py_DECREF(pyFrame);
```

Comment:
```suggestion
        Py_XDECREF(pyFrame);
```

Otherwise it can segfault if `_PyFrame_GetFrameObject` returned `NULL`